### DISCUSSION TITLE
Further details around database migration properties

### DISCUSSION THREAD

**Discussion by BR14Nx** [nextcloud/all-in-one#1290] [Questions] [answered]
I have been following the [complex migration](https://github.com/nextcloud/all-in-one/blob/main/migration.md#migrate-the-files-and-the-database) guide and have some questions about some details.  

About point **7.5**: Is the `local::/your/old/datadir/` entry the only one to change? I can see further entries about my former data directory path and my database is apparently quite big (>1GB), so I would like to know exactly what to change before I lay my hands onto something too unknown.

About point **9**: I assume `/var/lib/docker/volumes/nextcloud_aio_nextcloud_data/_data/*` is the standard data directory, right? First, I would suggest making a hint about that in the Readme, so it is more clear and coherent to everyone who are using a custom data path. And second, with ''restore the data directory of your former instance into the following directory'' it is meant to move the former data directory files in there, right?  

Last but not least, I was wondering how Nextcloud would handle the dump file. Is it just automatically importing the database and erase the former AIO database?  

For further information: My former database was a MariaDB and I went through all steps without any issues, but the AIO instance is exactly the same as before (and yes, I was also turning the containers off during the process). I know this migration is tricky and can increase the chance of errors, but I have barely any other choice since there are a lot of listed users and files on my Nextcloud.

**Comment by szaimen** [comment, marked as answer]
> About point **7.5**: Is the `local::/your/old/datadir/` entry the only one to change? I can see further entries about my former data directory path and my database is apparently quite big (>1GB), so I would like to know exactly what to change before I lay my hands onto something too unknown.

Yes, that should be the only needed change. What other entries do you have in your database that contain the path? Can you post them here?

> About point **9**: I assume `/var/lib/docker/volumes/nextcloud_aio_nextcloud_data/_data/*` is the standard data directory, right? First, I would suggest making a hint about that in the Readme, so it is more clear and coherent to everyone who are using a custom data path. And second, with ''restore the data directory of your former instance into the following directory'' it is meant to move the former data directory files in there, right?

Yes, this is the standard directory. 

Can you open a PR with the suggestion? Simply make the edits here: https://github.com/nextcloud/all-in-one/edit/main/migration.md and submit a PR :)

> Last but not least, I was wondering how Nextcloud would handle the dump file. Is it just automatically importing the database and erase the former AIO database?

Yes, it should automatically import the dump file upon starting the database container if this file is not present: `/var/lib/docker/volumes/nextcloud_aio_database_dump/_data/initial-cleanup-done`

**Reply by BR14Nx** [reply]
Nevermind, the datadirectory entry is only present once. Nevertheless I can find entries e.g. about some backup folder which changed (though might not be relevant due to new Borg Backups). For example: My previous nextcloud instance had a backup in `/var/nextcloud/nxtcld/backup`. Now the new datadirectory and backup is located in `/var/aionextcloud/` on another server. Should I keep the folder names the same?

To give some insight:
```
351     local::/var/nextcloud/nxtcld/backup/    0       1652685948
352     local::nextcloud/nxtcld/backup/ 0       1652369800
353     local::/nextcloud/nxtcld/backup/        0       1652369803
354     local::aawdwa/  0       1652371289
355     local::/var/nextcloud/  0       1652371677
356     local::/home/backup/    0       1652371801
357     local::var/nextcloud/nxtcld/backup/     0       1652371822
359     local::/var/www/html/backup/    1       1652692644
360     local::/var/www/html/backup/tmp_data/   0       1652692623
363     local::/var/backup/points/      1       1653918395
364     local::/var/backup/tmpdata/     0       1655297144
365     local::/var/backup/rpoints/     0       1655297144
```

**Reply by szaimen** [reply]
Is this from the [backup](https://apps.nextcloud.com/apps/backup) app? If yes you should ask your question [there](https://github.com/nextcloud/backup/issues)
